org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/typesystem/internal/LogicalContainerAwareReentrantTypeResolver.java

```diff
@@ -1011,10 +1011,12 @@ protected IFeatureScopeSession addExtensionFieldsToMemberSession(
 				}
 			}
 			// traverse the type hierarchy to create the feature scope sessions
-			JvmTypeReference superType = getExtendedClass(type);
 			IFeatureScopeSession result = featureScopeSession;
-			if (superType != null) {
-				result = addExtensionFieldsToMemberSession(resolvedTypes, featureScopeSession, (JvmDeclaredType) superType.getType(), thisFeature, seenNames, seenTypes);
+			for (JvmTypeReference ifx : type.getSuperTypes()) {
```

Comment:
is there a better way? do we need sorting?

Reply:
Yes, we'll need some sorting here to get a stable or understandable precedence. Now that we have to consider a graph due to multiple interface inheritance and also need to support duplicates that are *allowed* in that graph (following the superclass alone would only ever see dups for invalid hierarchies), a predictable order must be used.

org.eclipse.xtext.xbase.typesystem.override.ResolvedFeatures.computeAllOperationsFromSortedSuperTypes(JvmDeclaredType, Multimap<String, AbstractResolvedOperation>) might be a good starting point.

Reply:
arggg guess ill postpone until i find time to digg into